tuna/driver_base.py

```diff
@@ -141,36 +159,34 @@ def compose_input_t(self):
 
     return i_dict
 
+  def decompose_input_t(self, db_obj):
+    """Use input_tensor to assign local variables to build driver cmd """
+    #pylint: disable=attribute-defined-outside-init
+
+    self.set_cmd(db_obj.input_t.data_type)
+    self.num_dims = db_obj.input_t.num_dims
+    self.in_layout = db_obj.input_t.layout
+
+    if self.in_layout == 'NCHW':
+      self.in_channels = db_obj.input_t.dim1
+      self.in_d = db_obj.input_t.dim2
+      self.in_h = db_obj.input_t.dim3
+      self.in_w = db_obj.input_t.dim4
+    elif self.in_layout == 'NHWC':
+      self.in_d = db_obj.input_t.dim1
+      self.in_h = db_obj.input_t.dim2
+      self.in_w = db_obj.input_t.dim3
+      self.in_channels = db_obj.input_t.dim4
+
+    return True
+
   def get_weight_t_id(self):
     """Build 1 row in tensor table based on layout from fds param
      Details are mapped in metadata LAYOUT"""
     w_dict = self.compose_weight_t()
 
     return self.insert_tensor(w_dict)
 
-  def compose_weight_t(self):
-    """Build weight_tensor"""
-    w_dict = {}
-    w_dict['data_type'] = TENSOR_PRECISION[self.cmd]
-    w_dict['num_dims'] = self.num_dims
-
```

Comment:
This is moved to driver_conv.py. Is this because it is convolution specific? Is compose_input_t shared by batchnorm?